feat(workspace): Implement workspace management system

undead2146 opened the pull request:
* Closes #42
* Merge after #62

This pull request introduces a complete workspace management system that enables users to create isolated game environments using multiple preparation strategies.

### Changes
1. **Core Models & Interfaces**: Establishes the data contracts for workspace management including `WorkspaceConfiguration`, `WorkspaceInfo`, `GameLaunchConfiguration`, and service interfaces `IWorkspaceManager`, `IFileOperationsService`, `IGameLauncher`, `IWorkspaceStrategy`, and `IWorkspaceValidator`.

2. **Strategy Pattern Implementation**: Implements four distinct workspace preparation strategies:
   - `FullCopyStrategy`: Creates complete copies of all game files
   - `SymlinkOnlyStrategy`: Creates symbolic links (requires admin rights on Windows)
   - `HardLinkStrategy`: Creates hard links where possible with fallback to copies 
   - `HybridCopySymlinkStrategy`: Copies essential files (executables, configs, small files) and symlinks large media files for balanced disk usage and compatibility

3. **Service Implementations**: Provides concrete implementations for all workspace services:
   - `WorkspaceManager`: Orchestrates workspace creation, manages metadata persistence using JSON serialization, and coordinates with strategies
   - `FileOperationsService`: Handles low-level file operations including `CopyFileAsync`, `CreateSymlinkAsync`, `CreateHardLinkAsync`, `VerifyFileHashAsync`, and `DownloadFileAsync` with progress tracking
   - `WorkspaceValidator`: Validates `WorkspaceConfiguration` objects and checks prerequisites like admin rights and disk space
   - `GameLauncher`: Launches games from prepared workspaces using `Process.Start()` and manages `GameProcessInfo`

4. **Infrastructure Integration**: Updates the manifest system to use the new `WorkspaceStrategy.HybridCopySymlink` default and integrates all services through `WorkspaceModule.AddWorkspaceServices()`.

### Details
The system uses the strategy pattern where `WorkspaceManager.PrepareWorkspaceAsync()` accepts a `WorkspaceConfiguration` and delegates to the appropriate `IWorkspaceStrategy` implementation based on `configuration.Strategy`. Each strategy inherits from `WorkspaceStrategyBase<T>` which provides common functionality like progress reporting, file validation, and workspace metadata management. The `FileOperationsService` abstracts platform-specific operations, using P/Invoke for Windows hard links and falling back gracefully on other platforms.

1. **`IWorkspaceManager` (The Orchestrator):** Central service that coordinates workspace creation, manages metadata persistence, and delegates preparation tasks to appropriate strategies.
2. **`IWorkspaceStrategy` Implementations (The Workers):** Four concrete strategies (`FullCopyStrategy`, `SymlinkOnlyStrategy`, `HardLinkStrategy`, `HybridCopySymlinkStrategy`) that handle different approaches to workspace preparation.
3. **`IFileOperationsService` (The Operations Layer):** Low-level service providing file system operations including copying, symbolic links, hard links, hash verification, and HTTP downloads.
4. **`IGameLauncher` (The Execution Layer):** Service responsible for launching games from prepared workspaces and managing game processes.
5. **`IWorkspaceValidator` (The Safety Layer):** Validation service ensuring workspace configurations are valid and prerequisites are met before preparation begins.

undead2146 commented:
My most recent commit refactors  [FileOperationsService](https://github.com/community-outpost/GenHub/commit/10684c3a028f97fc84c892024f076bf2a1e5ad4d#diff-ca03519bb3737c4e3a5b415e6f4619f53fbaaccbb6e058d64fbcb979c357c6cd) to implement the Downloadservice merged in #64 , along with a couple changes like removing the duplicate [DownloadProgress.cs](https://github.com/community-outpost/GenHub/commit/10684c3a028f97fc84c892024f076bf2a1e5ad4d#diff-3a876786bc750c3999020bc29a769ed87bcb4ba97ebcd6b47e4aa3cc256ed537) and using the one provided by the earlier merge.

roossienb reviewed (changes requested):
It is a lot to review, so I may have missed things.

undead2146 commented on GenHub/GenHub/Features/Workspace/WorkspaceManager.cs:
```diff
@@ -0,0 +1,164 @@
+{
+    private readonly IEnumerable<IWorkspaceStrategy> _strategies = strategies;
+    private readonly ILogger<WorkspaceManager> _logger = logger;
+    private readonly string _workspaceMetadataPath = Path.Combine(
```
Done, I also made `SaveAllWorkspacesAsync()` static

undead2146 commented on GenHub/GenHub/Features/Workspace/FileOperationsService.cs:
```diff
@@ -0,0 +1,307 @@
+    {
+        try
+        {
+            var directory = Path.GetDirectoryName(linkPath);
+            if (!string.IsNullOrEmpty(directory))
+            {
+                Directory.CreateDirectory(directory);
```
I added the following utility methods. 
```csharp
    private static void EnsureDirectoryExists(string filePath)
    {
        var directory = Path.GetDirectoryName(filePath);
        if (!string.IsNullOrEmpty(directory))
        {
            Directory.CreateDirectory(directory);
        }
    }

    private static void DeleteFileIfExists(string filePath)
    {
        if (File.Exists(filePath))
        {
            File.Delete(filePath);
        }
    }
```

undead2146 commented on GenHub/GenHub/Features/Workspace/Strategies/HybridCopySymlinkStrategy.cs:
```diff
@@ -0,0 +1,202 @@
+                catch (Exception ex)
+                {
+                    var operation = isEssential ? "copy" : "create symlink for";
+                    Logger.LogError(
```
I have added the following method to the base class so that each strategy ends up using this method.
```csharp
    protected void CleanupWorkspaceOnFailure(string workspacePath)
    {
        try
        {
            if (FileOperationsService.DeleteDirectoryIfExists(workspacePath))
            {
                Logger.LogDebug("Successfully cleaned up workspace directory after failure: {WorkspacePath}", workspacePath);
            }
            else
            {
                Logger.LogWarning("Workspace directory did not exist or could not be deleted: {WorkspacePath}", workspacePath);
            }
        }
        catch (Exception cleanupEx)
        {
            Logger.LogWarning(cleanupEx, "Failed to cleanup workspace directory after failure: {WorkspacePath}", workspacePath);
        }
    }
```

undead2146 commented:
My most recent commit makes the following changes;

1. **DRY’ing up file‐ops**  
   - In `FileOperationsService` introduced three public helpers:  
     • `EnsureDirectoryExists(string path)`  
     • `DeleteFileIfExists(string path)`  
     • `DeleteDirectoryIfExists(string path)`  
   - Replaced all inline directory-creation & “if exists then delete” blocks with calls to those helpers.

2. **Centralized workspace-cleanup**  
   - In `WorkspaceStrategyBase<T>` added a protected `CleanupWorkspaceOnFailure(string workspacePath)` method  
   - All four strategies (`FullCopy`, `HardLink`, `HybridCopySymlink`, `SymlinkOnly`) now call that single cleanup helper instead of duplicating delete logic.

3. **Windows-only hard-link code moved**  
   - Removed `DllImport("kernel32.dll")` and P/Invoke from the shared library.  
   - Created `WindowsFileOperationsService` in the Windows project, decorated `IFileOperationsService`, and put the P/Invoke there.

4. **Updated dependency injection**  
   - `WorkspaceModule` (Core) registers only the shared services and strategies.  
   - In the Windows-specific DI module, injected `IFileOperationsService` with `WindowsFileOperationsService` to supply the hard-link implementation at runtime.

roossienb reviewed (approved):
Ok
